Field history tracking support

gnemiq opened the pull request:
#### Checklist
All items have to be completed before a PR is merged

- [x] Adhere to [Contribution Guidelines](https://docs.dxatscale.io/about-us/contributing-to-dx-scale)
- [ ] Updates to Decision Records considered?
- [ ] Updates to documentation at [DX@Scale Guide](https://github.com/dxatscale/dxatscale-guide) considered?
- [ ] Tested changes?
- [ ] Unit Tests new and existing passing locally?

azlam-abdulsalam committed:
Merge develop to main following May 22 (#985)

* revert(coverage): Fail Package validation if low coverage (#941)

Revert back to normal behaviour, as coverage calculation is stable with retries

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@19.0.11
 - @dxatscale/sfp-cli@2.0.11
 - @dxatscale/sfpowerscripts@13.0.13

* chore(release-title): update release title to May22

* feat(fast-feedback): experimental mode for faster feedback during validate (#945)

* feat(fast-feedback): experimental mode for faster feedback during validate

experimental feature: Add option which would help validate command to provide validation using

- Deploy only changed components of a changed package
- Trigger only testclasses for impacted classes
- Skip coverage validation
- Skip package dependencies installation

Utilize a hidden flag 'fastfeedback', this feature may be revoked at any time and should be considered unstable

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@19.1.0
 - @dxatscale/sfp-cli@2.0.12
 - @dxatscale/sfpowerscripts@13.1.0

* fix(validateimpl): fast feed back testing to respect testMode set in package

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts@13.1.1

* fix(apex-tests): ensure retry is available for all types of test modes

* fix(validate): fix headers being incorrectly displayed when there is no impacted test classes

* refactor(validate): use constants declared in core lib instead of hardcoding

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@19.1.1
 - @dxatscale/sfp-cli@2.0.13
 - @dxatscale/sfpowerscripts@13.1.2

* chore(dependencies): update @salesforce/core to 2.36.0

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@19.1.2
 - @dxatscale/sfp-cli@2.0.14
 - @dxatscale/sfpowerscripts@13.1.3

* fix(impacted-apex): add further conditions to impacted test class

- add test classes when it is changed
 -add all test classes to invalidated, when security model
changes

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@19.1.3
 - @dxatscale/sfp-cli@2.0.15
 - @dxatscale/sfpowerscripts@13.1.4

* fix(impacted-apex): use array concat instead of push, as it pushes whole array as a new element

* feat(validateimpl): add a env variable to support debugging

provide an env variable 'SFPOWERSCRIPTS_DEBUG_PREFETCHED_SCRATCHORG' to support during debugging
fetch. This allows to skip a new org from fetched from the pool

* fix(install-source): remove warning around selective, as its currently only used during fastfeedback

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@19.1.4
 - @dxatscale/sfp-cli@2.0.16
 - @dxatscale/sfpowerscripts@13.1.5

* refactor(trigger-apex-tests): converge writing junit into write block and remove redundant write

* feat(docker): add apex-link to docker image

* refactor(apex-tests): refactor to make it readable and fix related bugs

Refactor trigger logic to make it readable, fix the following issues as well
 1. Handle Tests with
multiple results
 2. Skip retrieving coverage when not requested \3.Split into smaller functions

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@19.1.5
 - @dxatscale/sfp-cli@2.0.17
 - @dxatscale/sfpowerscripts@13.1.6

* refactor(fast-feedback): move fast feedback from using env variable to a deployment type

* feat(fast-feedback): add fast feedback support in ValidateAgainstOrg

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@19.2.0
 - @dxatscale/sfp-cli@2.0.18
 - @dxatscale/sfpowerscripts@13.2.0

* fix(scripts): fix paths to pre and post deployment scripts when executed without repository (#954)

scripts paths is always pointed to one in the repository, rather than in the artifacts. This started
surfacing as the deploy impl generates a version of source package again from the provided artifact
(this is being refactored).

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@19.2.1
 - @dxatscale/sfp-cli@2.0.19
 - @dxatscale/sfpowerscripts@13.2.1

* fix(scripts): fix the incorrect fix to paths to pre and post

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@19.2.2
 - @dxatscale/sfp-cli@2.0.20
 - @dxatscale/sfpowerscripts@13.2.2

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@19.2.3
 - @dxatscale/sfp-cli@2.0.21
 - @dxatscale/sfpowerscripts@13.2.3

* fix: predeployment script path

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@19.2.4
 - @dxatscale/sfp-cli@2.0.22
 - @dxatscale/sfpowerscripts@13.2.4

* fix: upgrade semver from 7.3.5 to 7.3.6 (#956)

Snyk has created this PR to upgrade semver from 7.3.5 to 7.3.6.

See this package in npm:
https://www.npmjs.com/package/semver

See this project in Snyk:
https://app.snyk.io/org/azlam-abdulsalam/project/05b6b6d2-ba3e-421a-b920-224a7c9e0896?utm_source=github&utm_medium=referral&page=upgrade-pr

* fix: upgrade semver from 7.3.5 to 7.3.6 (#956)

Snyk has created this PR to upgrade semver from 7.3.5 to 7.3.6.

See this package in npm:
https://www.npmjs.com/package/semver

See this project in Snyk:
https://app.snyk.io/org/azlam-abdulsalam/project/05b6b6d2-ba3e-421a-b920-224a7c9e0896?utm_source=github&utm_medium=referral&page=upgrade-pr

* refactor/sfpPackage (#964)

* refactor(sfppackage): refactor to use sfppackage

* refactor(fast-feedback): migrate fastfeedback to build as opposed to validate building after success

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@20.0.0
 - @dxatscale/sfp-cli@2.1.0
 - @dxatscale/sfpowerscripts@13.3.0

* feat(deploy): add additional log with trace level

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts@13.3.1

* fix(artifacts): remove temp directories before packaging artifact

* fix(publish): fix issue when publishing artifacts bigger than 5 MB

fix #967

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@20.0.1
 - @dxatscale/sfp-cli@2.1.1
 - @dxatscale/sfpowerscripts@13.3.2

* fix(deploy): fix crash when skipIfInstall is used with deploy

* fix(test-options): fix lodash import

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@20.0.2
 - @dxatscale/sfp-cli@2.1.2
 - @dxatscale/sfpowerscripts@13.3.3

* fix(changelog): fix crash in changelog due to incorrect usage of sourceVersionFrom

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@20.0.3
 - @dxatscale/sfp-cli@2.1.3
 - @dxatscale/sfpowerscripts@13.3.4

* fix(package-test-coverage): fix matching of package type

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@20.0.4
 - @dxatscale/sfp-cli@2.1.4
 - @dxatscale/sfpowerscripts@13.3.5

* refactor(packagetype): move to an enum for packageType

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@21.0.0
 - @dxatscale/sfp-cli@2.2.0
 - @dxatscale/sfpowerscripts@13.4.0

* fix(package-type): fix issues with overriding packageType during validate

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@21.0.1
 - @dxatscale/sfp-cli@2.2.1
 - @dxatscale/sfpowerscripts@13.4.1

* fix(deploy): override package type in validate mode

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts@13.4.2

* fix(sfpackagebuilder): ignore diff computation on any error

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@21.0.2
 - @dxatscale/sfp-cli@2.2.2
 - @dxatscale/sfpowerscripts@13.4.3

* [CodeFactor] Apply fixes to commit cea880f (#969)

Co-authored-by: codefactor-io <support@codefactor.io>

* fix(fast-feedback): fix multiple issues with package diff creation

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@21.0.3
 - @dxatscale/sfp-cli@2.2.3
 - @dxatscale/sfpowerscripts@13.4.4

* refactor(sfppackagebuilder): remove usage of ChangeComponent and rather use componentSet directly

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@21.0.4
 - @dxatscale/sfp-cli@2.2.4
 - @dxatscale/sfpowerscripts@13.4.5

* fix(install-source): skip reconcile if no profiles are found in diff package

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@21.0.5
 - @dxatscale/sfp-cli@2.2.5
 - @dxatscale/sfpowerscripts@13.4.6

* fix(changedcomponents): revert to an earlier version,as its no longer used by fast feedback

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@21.0.6
 - @dxatscale/sfp-cli@2.2.6
 - @dxatscale/sfpowerscripts@13.4.7

* chore(dependencies): update @salesforce/core to 2.36.1

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@21.0.7
 - @dxatscale/sfp-cli@2.2.7
 - @dxatscale/sfpowerscripts@13.4.8

* chore(dependencies): update marked to 4.0.14

* chore(dependencies): update semver to 7.3.6

* chore(dependencies): update @oclif/plugin-help to 5.1.12

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@21.0.8
 - @dxatscale/sfp-cli@2.2.8
 - @dxatscale/sfpowerscripts@13.4.9

* chore(dependencies): update source-deploy-retrieve to 5.12.13

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@21.0.9
 - @dxatscale/sfp-cli@2.2.9
 - @dxatscale/sfpowerscripts@13.4.10

* fix(apex-tests): do not replace coverage when coverage is not requested

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@21.0.10
 - @dxatscale/sfp-cli@2.2.10
 - @dxatscale/sfpowerscripts@13.4.11

* feat(validate): do not delete scratch org when there is no package to be built (#972)

* feat(validate): do not delete scratch org when there is no package to be built

fix #971

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@21.0.11
 - @dxatscale/sfp-cli@2.2.11
 - @dxatscale/sfpowerscripts@13.4.12

* chore(dependencies): update sfdc-soup to 18.2.1

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@21.0.12
 - @dxatscale/sfp-cli@2.2.12
 - @dxatscale/sfpowerscripts@13.4.13

* fix(fast-feedback): fix and refactor impacted test class selection

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts@13.4.14

* fix(log): fix time displayed as well as packages desplaying undefined keys

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@21.0.13
 - @dxatscale/sfp-cli@2.2.13
 - @dxatscale/sfpowerscripts@13.4.15

* fix(scratchorg-info-assigner): fix incorrect use of async-retry

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@21.0.14
 - @dxatscale/sfp-cli@2.2.14
 - @dxatscale/sfpowerscripts@13.4.16

* fix(get-formatted-time): fix get formatted time to use milliseconds

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@21.0.15
 - @dxatscale/sfp-cli@2.2.15
 - @dxatscale/sfpowerscripts@13.4.17

* fix(pool): fix issues while returning scratch org to pool

the code was using incorrect Id of the scratch org.. this fixes to the correct id, However its is
still dependent on pre-requisite pool package

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@21.0.16
 - @dxatscale/sfp-cli@2.2.16
 - @dxatscale/sfpowerscripts@13.4.18

* fix(data-pkg): fix vlocity package being incorrectly returned as empty

* fix(install-source): packages are committed even when push fails

fix issue where packages are committed to the org even when push fails to deploy, as error was
incorrectly handled

* feat(fast-feedback): fast feedback to skip packages from validation when descriptor changes

validate will validate the full package when the descriptor of the package changes. Most often the
change is in a version number, and this triggers full deployment and testing, in fast feedback mode,
this could be skipped. This has a negative issue when additonal orchestrator parameters are added
like assignPermissionSets etc, this could be validated in a thorough mode

* feat(fast-feedback): unhide fast-feedback and display info in header

* feat(reconcile): enhance how reconcile is handled during fast feedback

* docs(dr): add a dr on fast feedback

* chore(sfpowerkit): update sfpowerkit to 4.2.7

* fix(alias): fix aliasified packages using username instead of alias

Fix aliasified packages t use aliases instead of username. Refactor to utilize sfpOrg as opoosed to
passing targetusername

* fix(install-package): fix import casing

* chore(dockerfile): update sfpowerkit to 4.2.8

* fix(creation-metrics): fix package creation metrics reporting

package metrics during creation is not sending the flag whether its used with build or quickbuild,
added an additional metric to deprecate package.elapsed.time in the future

* chore(dependencies): update @salesforce/core to 2.36.3

* fix(dependencies): update import of core to same version in core

* fix(package-lock): sync package lock across

* Add chunking to ApexClassFetcher, ApexTriggerFetcher and ACCG Fetcher (#953)

* Add chunking to ApexClassFetcher, ApexTriggerFetcher and ACCG Fetcher

* Fix return values

Co-authored-by: Azlam <43767972+azlam-abdulsalam@users.noreply.github.com>

* fix(fast-feedback): fix All tests  getting triggered when invalidated test class was not found

* Update 002-fast-feedback.md

Fixed typos and minor edits to wording for better flow.

* fix(publish): create tags when atleast one package is published

fix a bug when a tag when the same package would have been published

* fix(fast-feedback): add warning message when tests are skipped due to a full package being deployed

* fix(install-package): fix logs to show 'default' when aliasifed folder is not found

* fix(fast-feedback): add additional logs about selective components utilized in fast feedback

* fix(validate): add error to display when validate fails for anyother reason

* fix(validate): remove erroneous json stringify on error message

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts@13.6.13

* chore(dependencies): update simple-git to 3.7.1

* chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@21.3.3
 - @dxatscale/sfp-cli@2.3.3
 - @dxatscale/sfpowerscripts@13.6.14

* fix(git): use command to fetch git remote instead of simple-git

* fix(git): fix git related issues on docker

- Rollback change made earlier to simple-git
  - Use pinned version of heroku docker image as this
version seems not to have the issue. This will be analyzed in a future release

Co-authored-by: Alan Ly <alan.ly@accenture.com>
Co-authored-by: Snyk bot <snyk-bot@snyk.io>
Co-authored-by: codefactor-io <support@codefactor.io>
Co-authored-by: Alan Ly <36793262+aly76@users.noreply.github.com>
Co-authored-by: Vu Ha <59901319+vuha-acn@users.noreply.github.com>

vuha-acn committed:
Updated validate cmd help manuals for fastfeedback (#986)

vuha-acn committed:
Update validateAgainstOrg command help docs (#987)

Commit:
Fetch changes from develop

azlam-abdulsalam committed:
chore(cicd): add a cherrypick action

Add an action to cherrypick changes main to develop

Commit:
Merge updates from develop

azlam-abdulsalam committed:
chore(cicd): Remove auto trigger on main

[ci skip]

dieffrei committed:
fix(install-package): always check if there is any permissionset group updating before package install (#1002)

fixes #1000 always wait a package installation if there is any ongoing PSG calculation

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@21.3.8
 - @dxatscale/sfp-cli@2.3.8
 - @dxatscale/sfpowerscripts@13.6.22

Commit:
Merge June 22 to main

vuha-acn committed:
Updated links from Docker to GHCR

Commit:
fix(docker-image): fix docker image title to June 22

Commit:
[ci skip] Merge branch 'develop'

azlam-abdulsalam committed:
fix(sfdx-cli): update sfdx-cli to 7.156.1 (#1073)

Due to #1576 and #1577 in forcedotcom-cli, an update to cli is warranted

fix #1072

Commit:
docs(licenses): update third party notices info

azlam-abdulsalam committed:
fix(deploy): fix skipTesting not respected during validate (#1078)

* fix(deploy): fix skipTesting not respected during validate

skipTesting was not respected during validate, when optimized deployment was set to false

fix #1075

* fix(deploy): fix incorrect variable

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@23.4.6
 - @dxatscale/sfpowerscripts@15.3.3

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@23.4.7
 - @dxatscale/sfpowerscripts@15.3.4

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@23.4.8
 - @dxatscale/sfpowerscripts@15.3.5

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@23.4.9
 - @dxatscale/sfpowerscripts@15.3.6

Commit:
18.1.3

Commit:
fix(cicd): remove caching on review jobs

azlam-abdulsalam committed:
Downgrade sfdx data plugin to 2.0.0 (#1085)

The data plugin is throwing an issue with force:data:soql:query resulting in pre/post scripts failing

Schuchie committed:
Issue/1086 (#1087)

* fix(releasedefinitiongenerator): fixed typo in flags usage of release definition generator command

* fix(releasedefinitiongenerator): allow different branch name on changelog parameter usage

Co-authored-by: Azlam <43767972+azlam-abdulsalam@users.noreply.github.com>
Co-authored-by: zhebinliu <85538531+zhebinliu@users.noreply.github.com>

Commit:
fix(deps): update @salesforce/source-deploy-retrieve to 6.2.7

Commit:
chore(package-lock): update package locs

zhebinliu committed:
fix: 🐛 (apextest)add extra checks on if condition (#1100)

check whether the parent variables are undefined when handling version
number

✅ Closes: #1084

azlam-abdulsalam committed:
fix(docs): fix number on dr

[ci skip]

azlam-abdulsalam committed:
Fix broken link

[ci skip]

Commit:
August 22 Release

Chiqqn committed:
fix(tests): Duplicate checking apexClass name together with methodNam… (#1112)

fix(tests): Duplicate checking apexClass name together with methodName in removeDuplicateTestListing

Co-authored-by: Dominik Skoczylas <dominik.skoczylas@expereo.com>

Commit:
[ci skip] Merge September 22 release

Commit:
fix(releasedefintion): fix unpredicatable behaviour in releasedefn generator

git checkout should be synchronous, the usage was missing in await

fix #1127

dieffrei committed:
feat(metrics): Set target_org tag for SO generic metrics (#1143)

Add additional tag for devhub when projects use multiple metrics

Co-authored-by: Diéffrei Tiepo de Quadros <dieffrei.quadros@expereo.com>

dependabot[bot] committed:
chore(deps): bump @xmldom/xmldom from 0.8.2 to 0.8.5 in /packages/core (#1141)

Bumps [@xmldom/xmldom](https://github.com/xmldom/xmldom) from 0.8.2 to 0.8.5.
- [Release notes](https://github.com/xmldom/xmldom/releases)
- [Changelog](https://github.com/xmldom/xmldom/blob/master/CHANGELOG.md)
- [Commits](https://github.com/xmldom/xmldom/compare/0.8.2...0.8.5)

---
updated-dependencies:
- dependency-name: "@xmldom/xmldom"
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Azlam <43767972+azlam-abdulsalam@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump loader-utils from 1.4.0 to 1.4.2 (#1147)

Bumps [loader-utils](https://github.com/webpack/loader-utils) from 1.4.0 to 1.4.2.
- [Release notes](https://github.com/webpack/loader-utils/releases)
- [Changelog](https://github.com/webpack/loader-utils/blob/v1.4.2/CHANGELOG.md)
- [Commits](https://github.com/webpack/loader-utils/compare/v1.4.0...v1.4.2)

---
updated-dependencies:
- dependency-name: loader-utils
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

gnemiq committed:
Field history tracking resolution DR (#1134)

* Field history tracking resolution DR

* update context

* update text regarding fht

* add issues/challenges associated with the solution

* update diagram

* update diagram

* update diagram

* update diagram

* update diagram

* Add additional issues/challenge and typo fixes

* updated dr with additional changes

Co-authored-by: DIR\meng.qi <meng.b.qi@accenture.com>
Co-authored-by: Azlam <43767972+azlam-abdulsalam@users.noreply.github.com>
Co-authored-by: Vu Ha <59901319+vuha-acn@users.noreply.github.com>

dieffrei committed:
fix(prepeareorg): fix metrics with additional tags

* chore(pool): change name of single job metric and add pooltag

* fix(prepeareorg): fix metrics with additional tags

fix individual job metrics with pool info

Co-authored-by: Diéffrei Tiepo de Quadros <dieffrei.quadros@expereo.com>
Co-authored-by: Azlam <43767972+azlam-abdulsalam@users.noreply.github.com>

Commit:
Merge Nov 22 release into main

[ci skip]

Commit:
chore(deps): update core to 3.32.8

Commit:
fix(publish): fix publish adding an additional '@'

Changes in @salesforce/command has led to ignoring the removal of '@' symbol added by a user during
the parsing of the flag. This has led subsequent code to add '@' again, as '@' is mandatory for
scopde package and users may ignore adding '@' to the scope. This was clearly the established
behaviour with publish command. This fix reintroduces previous behaviours

issue not created in github, but refered in slack as
https://dxatscale.slack.com/archives/CN9N54GF5/p1670321226990929

gnemiq committed:
Field history tracking support

gnemiq committed:
Merge branch 'develop' into feat/fht-support

gnemiq committed:
update variable name

azlam-abdulsalam commented on packages/core/src/package/packageInstallers/InstallPackage.ts:
```diff
@@ -242,6 +245,106 @@ export abstract class InstallPackage {
         }
+
+        //run the post deployment fht enabling process
+        let modifiedComponentSet = await FHTPostDeploymentGenerator.generateFHTEnabledComponents(this.sfpPackage, this.connection, this.logger);
```
Lets move this into a function for more readability

attilah committed:
chore(logs): fix typos and double spaces in strings (#1171)

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@33.2.8
 - @dxatscale/sfpowerscripts@20.2.7

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@33.2.9
 - @dxatscale/forcemula@0.1.5
 - @dxatscale/sfdc-soup@18.2.5
 - @dxatscale/sfpconsole@1.0.3
 - @dxatscale/sfp-logger@1.1.2
 - @dxatscale/sfpowerscripts@20.2.8

Commit:
refactor(cicd): remove alpha environment from cicd

Commit:
refactor(cicd): add run name to workflow

Commit:
refactor(cicd): remove concurrency on build

Commit:
fix(validateagainstorg): validateAgainstOrg is not installing exsternal deps in individual mode

validateAgainstOrg needs access to devhub and should install  external dependencies when a devhub is

Commit:
fix(cicd): fix incorrect stage usage

Commit:
fix(cicd): fix incorrect stage usage

Commit:
fix(cicd): fix incorrect indentation and stage name

Commit:
fix(createunlocked): unlocked creation not respecting config path

definitionFile param used by the packaging lib expects fully resolved path and doesnt operate in the
context of sfProject

closes #1176

Commit:
refactor(cicd): add new envs

Commit:
fix(cicd): remove conditional on prod

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@33.2.10
 - @dxatscale/sfpowerscripts@20.2.10

Commit:
fix(unlockedpackage): fix errors not displayed correctly

errors was not displayed when there is a failure

closes #1177

Commit:
fix(cicd): fix deps of alpha build

Commit:
fix(cicd): remove concurrency group

gnemiq committed:
udpate fht class names & logic

gnemiq committed:
resolve conflict

gnemiq committed:
Merge branch 'develop' into feat/fht-support

gnemiq committed:
resolve conflicts

gnemiq committed:
remove release.yml

gnemiq committed:
resovle test errors

Commit:
revert changes to release.yml

Commit:
refactor(fht): generalize fht enabler
